drivers/usb/device/usb_dc_stm32.c

```diff
@@ -89,6 +89,13 @@ static const struct gpio_dt_spec ulpi_reset =
 	GPIO_DT_SPEC_GET_OR(DT_PHANDLE(DT_INST(0, st_stm32_otghs), phys), reset_gpios, {0});
 #endif
 
+#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(USB_OTG_HS_EMB_PHYC) || \
+	defined(USB_OTG_HS_EMB_PHY)
+#define HIGH_SPEED             USB_OTG_SPEED_HIGH_IN_FULL
+#else
+#define HIGH_SPEED             USB_OTG_SPEED_HIGH
+#endif
+#define FULL_SPEED             USB_OTG_SPEED_FULL
```

Comment:
Nit: spacing to make end of `#ifdef` block more visible
```suggestion
#endif

#define FULL_SPEED             USB_OTG_SPEED_FULL
```